submittal/src/main/java/com/kyriba/submittal/api/ConfirmationController.java

```diff
@@ -0,0 +1,48 @@
+package com.kyriba.submittal.api;
+
+import com.kyriba.submittal.service.SpeakerService;
+import lombok.Builder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;
+
+
+/**
+ * @author M-ABL
+ */
+@RestController
+@RequestMapping(value = "/api/v1/confirm",
+    consumes = APPLICATION_JSON_UTF8_VALUE,
+    produces = APPLICATION_JSON_UTF8_VALUE)
+public class ConfirmationController
+{
+  private final SpeakerService speakerService;
+
+
+  @Autowired
+  ConfirmationController(final SpeakerService speakerService)
+  {
+    this.speakerService = speakerService;
+  }
+
+
+  @GetMapping(path = "/registration")
+  public RegistrationConfirmedResponse confirmRegistration(@RequestParam final String key)
+  {
+    return RegistrationConfirmedResponse.builder()
+        .speakerId(speakerService.finishRegistration(key).getSpeakerId())
+        .build();
+    // TODO: what's next? Invoke Conference MS directly in order to submit topics?
+  }
+
+
+  @Builder
+  private static class RegistrationConfirmedResponse

```

Comment:
Builder is useless for such DTO, this is Value Object

Reply:
please explain. I use the builder in order to set values and return as response

Reply:
Builder pattern is used when class has many fields and dynamic creation options are needed. With single param it is easier to use Value annotation and constructor for creation.